include/__generator.hpp

```diff
@@ -492,8 +500,31 @@ struct __generator_promise<generator<_Ref, _Value, _Alloc>, _ByteAllocator> fina
             std::coroutine_handle<__generator_promise>::from_promise(*this)
         };
     }
+
+    using __generator_promise_base<_Ref>::yield_value;
+
+    template <std::ranges::range _Rng>
+    typename __generator_promise_base<_Ref>::template __yield_sequence_awaiter<generator<_Ref, std::remove_cvref_t<_Ref>, _Alloc>>
+    yield_value(std::ranges::elements_of<_Rng> && __x) {
+        static_assert (!_ExplicitAllocator,
+        "This coroutine has an explicit allocator specified with std::allocator_arg so an allocator needs to be passed "
+        "explicitely to std::elements_of");
+        return [](auto && __rng) -> generator<_Ref, std::remove_cvref_t<_Ref>, _Alloc> {
```

Comment:
We have access to the `_Value` type here, can we use it rather than `remove_cvref_t<_Ref>` to avoid extra template instantiations?